product_tax_multicompany_default/models/product.py

```diff
@@ -0,0 +1,257 @@
+# Copyright 2017 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# Copyright 2018 Vicent Cubells - Tecnativa <vicent.cubells@tecnativa.com>
+# Copyright 2023 Eduardo de Miguel - Moduon <edu@moduon.team>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    divergent_company_taxes = fields.Boolean(
+        string="Has divergent cross-company taxes",
+        compute="_compute_divergent_company_taxes",
+        compute_sudo=True,
+        help=(
+            "Does this product have divergent cross-company taxes? "
+            "(Only for multi-company products)"
+        ),
+    )
+
+    @api.depends("company_id", "taxes_id", "supplier_taxes_id")
+    def _compute_divergent_company_taxes(self):
+        """Know if this product has divergent taxes across companies."""
+        for one in self:
+            one.divergent_company_taxes = False
+            # Skip single-company products
+            if one.company_id or len(one.env["res.company"].search([]).ids) <= 1:
+                continue
+            # A unique constraint in account.tax makes it impossible to have
+            # duplicated tax names by company
+            one.divergent_company_taxes = self._is_divergent_company_taxes(one, "taxes")
+            if one.divergent_company_taxes:
+                continue
+
+            one.divergent_company_taxes = self._is_divergent_company_taxes(
+                one, "purchase"
+            )
+            if one.divergent_company_taxes:
+                continue
+
+    def _is_divergent_company_taxes(self, current_product, tax_type):
+        """Returns true or false if there are differences in product taxes.
+
+        :param one: product to be checked
+        :param tax_type: can be 'taxes' or 'purchase'
+        """
```

Comment:
If you are only to use one product, make it available direclty from product. Otherwise, use  @api.model

Update parameters help


```suggestion
    def _is_divergent_company_taxes(self, tax_type):
        """Returns true or false if there are differences in product taxes.

        :param tax_type: can be 'taxes' or 'purchase'
        """
        self.ensure_one()
```